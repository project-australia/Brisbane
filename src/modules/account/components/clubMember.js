import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, ScrollView, Text, Alert } from 'react-native'
import { Navbar } from '../../shared/components/navbar'
import { FormOutlineButton } from '../../shared/components/buttons'
import { payWithPayPal } from '../../../services/paypal'
import { styles } from './styles/clubMember.style'

export class ClubMember extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    goBack: PropTypes.func.isRequired,
    logOut: PropTypes.func.isRequired,
    updateProfile: PropTypes.func.isRequired,
    buttonText: PropTypes.string.isRequired,
    user: PropTypes.object.isRequired,
    club: PropTypes.oneOf(['TEN', 'TWENTY', 'NONE']).isRequired,
    price: PropTypes.number.isRequired
  }

  static navigationOptions = {
    title: 'Join our club membership',
    header: null
  }

  show10club = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Get 10% More for books you sell</Text>
        <Text style={styles.text}>• Get 10% Off of purchases</Text>
        <Text style={styles.text}>• Get 10% Off of rentals</Text>
      </ScrollView>
    )
  }

  show20club = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Get 20% More for books you sell</Text>
        <Text style={styles.text}>• Get 20% Off of purchases</Text>
        <Text style={styles.text}>• Get 20% Off of rentals</Text>
      </ScrollView>
    )
  }

  showRep = () => {
    return (
      <ScrollView style={styles.wrapper}>
        <Text style={styles.text}>• Just sign people up to the app (friends, classmates, coworkes, teammates, etc.)</Text>
        <Text style={styles.text}>• When they buy, sell or rent you get a commiission</Text>
        <Text style={styles.text}>• If they become a rep too, you even get a commission from the people they sign up</Text>
        <Text style={styles.text}>• The more you sign up the better!</Text>
        <Text style={styles.text}>• It's so easy!</Text>
      </ScrollView>
    )
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <Navbar
          title={this.props.title}
          onBack={this.props.goBack}
          ignoreAndroidStatusBar
        />
        {this.props.club === 'NONE' && this.show10club()}
        {this.props.club === 'TEN' && this.show20club()}
        {this.props.club === 'TWENTY' && this.showRep()}
        {this.props.club !== 'TWENTY' && <FormOutlineButton
          title={this.props.buttonText}
          onPress={this.checkoutWithPaypal}
          style={styles.input}
        />
        }
      </View>
    )
  }
//   <View style={styles.buttonGroup}>
//   <FlatButton
//     secondary
//     containerStyle={styles.button}
//     title={this.props.buttonText}
//     onPress={this.checkoutWithPaypal}
//   />
// </View>
  checkoutWithPaypal = async () => {
    try {
      await payWithPayPal(
        this.props.price,
        this.props.title,
        this.onPayPalOnSuccess
      )
    } catch (error) {
      console.log('Paypal checkout failed', JSON.stringify(error))
      this.alert('Sorry. Request Failed')
    }
  }

  onPayPalOnSuccess = async paypalResponse => {
    const { user, club, updateProfile, navigate } = this.props
    await updateProfile(user.id, { club })
    this.alert('Successfully registered', navigate('Home'))
  }

  alert = (msg, press = () => {}) =>
    Alert.alert(
      this.props.title,
      msg,
      [
        {
          text: 'Ok',
          onPress: () => press
        }
      ],
      { cancelable: false }
    )
}
