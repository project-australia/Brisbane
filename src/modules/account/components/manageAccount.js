import React, { Component } from 'react'
import { View } from 'react-native'

export class ManageAccount extends Component {
  render () {
    return <View />
  }
}
