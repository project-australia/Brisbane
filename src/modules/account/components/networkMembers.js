import React from 'react'
import { ScrollView, Text, View } from 'react-native'

import { Navbar } from '../../shared/components/navbar'

import { styles } from './styles/networkMembers.style'

const renderMember = member => (
  <View key={member.id} style={styles.wrap}>
    <Text numberOfLines={1} style={styles.title}>
      {member.name}
    </Text>
    <Text numberOfLines={1} style={styles.subtitle}>
      {member.school}
    </Text>
    <Text numberOfLines={1} style={styles.subtitle}>
      {member.email}
    </Text>
  </View>
)

export const NetworkMembers = ({ navigateBack, network }) => {
  return (
    <View style={styles.container}>
      <Navbar title={'Network Members'} onBack={navigateBack} />
      <ScrollView>{network.map(renderMember)}</ScrollView>
    </View>
  )
}
