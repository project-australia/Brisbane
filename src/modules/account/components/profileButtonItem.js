import React from 'react'
import { View } from 'react-native'

import { MenuTitle } from '../../shared/components/menuTitle'
import { FlatButton } from '../../shared/components/buttons'

import { styles } from './styles/profileItem.style'

export const ProfileButtonItem = ({ title, buttonTitle, onPress, secondary }) => {
  return (
    <View style={styles.wrap}>
      <MenuTitle title={title} />
      <FlatButton
        secondary={secondary}
        title={buttonTitle}
        containerStyle={styles.itemBackground}
        onPress={onPress}
      />
    </View>
  )
}
