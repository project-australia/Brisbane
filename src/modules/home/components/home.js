import React from 'react'
import PropTypes from 'prop-types'
import { Button, Text, View } from 'react-native'

import { styles } from './styles/home.styles'
import { SellingBooks } from '../containers/sellingBooksContainer'
import { WalletBalanceAmount } from '../containers/walletBalanceContainer'
import { BookSearch } from './searchBook'
import { book } from '../propTypes/book'
import { HorizontalBookList } from './HorizontalBookList'

export class Home extends React.Component<{}> {
  static propTypes = {
    navigateToProfile: PropTypes.func.isRequired,
    navigateToScan: PropTypes.func.isRequired,
    navigateToShoppingBag: PropTypes.func.isRequired,
    searchBook: PropTypes.func.isRequired,
    recentlyAddedBooks: PropTypes.arrayOf(book).isRequired,
    featuredBooks: PropTypes.arrayOf(book).isRequired,
    userName: PropTypes.string
  }

  static defaultProps = {
    userName: 'Guest'
  }

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          {`Hi, ${this.props.userName}!`}
        </Text>
        <BookSearch
          onSubmit={this.props.searchBook}
          onScanPress={this.props.navigateToScan}
        />
        <Button
          title='PROFILE'
          onPress={this.props.navigateToProfile}
        />
        <Button
          title='SHOPPING BAG'
          onPress={this.props.navigateToShoppingBag}
        />
        <SellingBooks />
        <WalletBalanceAmount />
        <Text style={styles.welcome}>
          Recently added
        </Text>
        <HorizontalBookList
          books={this.props.recentlyAddedBooks}
        />
        <Text style={styles.welcome}>
          Recently added
        </Text>
        <HorizontalBookList
          books={this.props.recentlyAddedBooks}
        />
      </View>
    )
  }
}
