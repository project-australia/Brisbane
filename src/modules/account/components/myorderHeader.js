import React from 'react'
import { Text, View } from 'react-native'
import { styles } from './styles/networkMembers.style'

const property = ({ title, property }) => (
  <Text key={title} style={styles.propertyTitle}>
    {title}: <Text style={styles.propertyText}>{property}</Text>
  </Text>
)

export const MyOrderHeader = ({ title, properties }) => (
  <View style={[styles.wrap, styles.standardCardPadding]}>
    <Text style={styles.title}>{title}</Text>
    <View style={styles.textBlockWrap}>{properties.map(property)}</View>
  </View>
)
