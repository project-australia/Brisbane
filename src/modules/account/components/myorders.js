import React from 'react'
import { ScrollView, Text, View } from 'react-native'

import { Navbar } from '../../shared/components/navbar'

import { styles } from './styles/networkMembers.style'

const renderOrder = order => (
  <View key={order.id} style={styles.wrap}>
    <Text numberOfLines={1} style={styles.title}>
      {order.id}
    </Text>
    {order.items.map(book => (
      <Text key={book.id} numberOfLines={1} style={styles.title}>
        {book.title}
      </Text>
    ))}
  </View>
)

export const MyOrders = ({ navigateBack, orders }) => {
  return (
    <View style={styles.container}>
      <Navbar title={'My Orders'} onBack={navigateBack} />
      <ScrollView>{orders.map(renderOrder)}</ScrollView>
    </View>
  )
}
