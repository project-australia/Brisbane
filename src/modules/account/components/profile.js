import React from 'react'
import { ScrollView, View } from 'react-native'
import moment from 'moment'

import { ModalWithInputProfile } from '../../shared/components/modals/modalWithInputProfile'
import { Navbar } from '../../shared/components/navbar'
import { ProfileItem } from './profileItem'
import { ProfileItemAddress } from './profileItemAddress'
import { ScreenTitleAndSubtitle } from './screenTitleAndSubtitle'
import { ProfileRedirectItem } from './profileRedirectItem'
import { ProfileButtonItem } from './profileButtonItem'

import { styles } from './styles/profile.style'

const getSubscriptionText = subType => {
  switch (subType) {
    case 'TEN':
      return '10% off'
    case 'TWENTY':
      return '20% off'
    default:
      return 'Standard'
  }
}

const setUserNetworkNumber = network =>
  network && network.length > 0 && `${network.length} registered members`

export const Profile = ({
  onBackPress,
  user,
  modalTitle,
  navigateToNetwork,
  navigateToWallet,
  onLogoutPress,
  showEditModal,
  isEditModalOpen,
  onConfirmModal,
  onDismissModal
}) => {
  const rightIcon = [
    {
      name: 'pencil',
      onPress: () => showEditModal('Address')
    }
  ]
  return (
    <View style={styles.container}>
      <Navbar onBack={onBackPress} rightIcons={rightIcon} />
      <ScrollView>
        <ScreenTitleAndSubtitle
          title={user.name}
          subtitle={`${getSubscriptionText(user.subscription)} member`}
        />
        <ProfileItem
          title={'School'}
          data={user.school}
          emptyDataLabel={'Add school name'}
          onPress={() => showEditModal('School')}
        />
        <ProfileItem
          title={'Phone'}
          data={user.telephone}
          emptyDataLabel={'Add phone number'}
          onPress={() => showEditModal('Phone')}
        />
        <ProfileItem
          title={'Birth date'}
          data={moment(user.birthDate).format('MMM Do YYYY')}
          emptyDataLabel={'Add birth date'}
          onPress={() => showEditModal('Birth date')}
        />
        <ProfileItemAddress
          title={'Address'}
          data={user.address}
          emptyDataLabel={'Add address'}
        />
        <ProfileRedirectItem
          title={'Network'}
          data={setUserNetworkNumber(user.network)}
          onPress={navigateToNetwork}
        />
        <ProfileButtonItem
          secondary
          title={'My Wallet'}
          buttonTitle={'View my Wallet'}
          onPress={navigateToWallet}
        />
        <ProfileButtonItem buttonTitle={'Logout'} onPress={onLogoutPress} />
      </ScrollView>
      <ModalWithInputProfile
        visible={isEditModalOpen}
        title={modalTitle}
        user={user}
        onConfirm={onConfirmModal}
        onDismiss={onDismissModal}
      />
    </View>
  )
}
