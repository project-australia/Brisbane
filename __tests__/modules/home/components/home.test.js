import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Home } from '../../../../src/modules/home/components/home'

const props = {
  userName: 'Patricinha',
  navigateToProfile: () => null,
  navigateToShoppingBag: () => null
}

describe('<Home />', () => {
  xit('should have a snapshot', () => {
    const wrapper = shallow(<Home {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should have guest as default userName', () => {
    expect(Home.defaultProps.userName).toEqual('Guest')
  })
})
