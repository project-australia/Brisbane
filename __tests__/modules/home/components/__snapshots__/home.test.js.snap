// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should have a snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Hi, Patricinha!
  </Text>
  <Button
    onPress={[Function]}
    title="PROFILE"
  />
  <Button
    onPress={[Function]}
    title="SHOPPING BAG"
  />
</View>
`;
